openedx/core/djangoapps/discussions/views.py

```diff
@@ -181,25 +204,26 @@ def post(self, request, course_key_string: str, **_kwargs) -> Response:
 
         TODO: Should we cleanup orphaned LTI config when swapping to cs_comments_service?
         """
-        course_key = self._validate_course_key(course_key_string)
+        course_key = _validate_course_key(course_key_string)
         configuration = DiscussionsConfiguration.get(course_key)
         serializer = self.Serializer(configuration, data=request.data, partial=True)
         if serializer.is_valid(raise_exception=True):
             serializer.save()
         return Response(serializer.data)
 
-    def _validate_course_key(self, course_key_string: str) -> CourseKey:
-        """
-        Validate and parse a course_key string, if supported
-        """
-        try:
-            course_key = CourseKey.from_string(course_key_string)
-        except InvalidKeyError as error:
-            raise serializers.ValidationError(
-                f"{course_key_string} is not a valid CourseKey"
-            ) from error
-        if course_key.deprecated:
-            raise serializers.ValidationError(
-                'Deprecated CourseKeys (Org/Course/Run) are not supported.'
-            )
-        return course_key
+
+def _validate_course_key(course_key_string: str) -> CourseKey:
+    """
+    Validate and parse a course_key string, if supported
+    """
+    try:
+        course_key = CourseKey.from_string(course_key_string)
+    except InvalidKeyError as error:
+        raise serializers.ValidationError(
+            f"{course_key_string} is not a valid CourseKey"
+        ) from error
+    if course_key.deprecated:
+        raise serializers.ValidationError(
+            'Deprecated CourseKeys (Org/Course/Run) are not supported.'
+        )
+    return course_key
```

Comment:
Maybe these functions too could be moved to a shared location and imported both here and in `learning_sequences/views.py`?

Reply:
Agree, see above.